typo: fix missing backtick (#20202)
cloudflare/cloudflare-docs: src/content/docs/agents/concepts/calling-llms.mdx

---
title: Calling LLMs
pcx_content_type: concept
sidebar:
  order: 6

---

import { Render } from "~/components";

> ### Understanding LLM Providers and Model Types

Reviewer: ```suggestion
### Understanding LLM providers and model types
```

Different LLM providers offer models optimized for specific types of tasks. When building AI systems, choosing the right model is crucial for both performance and cost efficiency.

> #### Reasoning Models

Reviewer: ```suggestion
#### Reasoning Models

```

> Models like OpenAI's o1, Anthropic's Claude, and DeepSeek's R1 are particularly well-suited for complex reasoning tasks. These models excel at:

Reviewer: ```suggestion
Models like OpenAI's o1, Anthropic's Claude, and DeepSeek's R1 are particularly well-suited for complex reasoning tasks. These models excel at:

```

- Breaking down problems into steps
- Following complex instructions
- Maintaining context across long conversations
- Generating code and technical content

For example, when implementing a travel booking system, you might use a reasoning model to analyze travel requirements and generate appropriate booking strategies.

> #### Instruction Models

Reviewer: ```suggestion
#### Instruction Models

```

> Models like GPT-4 and Claude Instant are optimized for following straightforward instructions efficiently. They work well for:

Reviewer: ```suggestion
Models like GPT-4 and Claude Instant are optimized for efficiently following straightforward instructions. They work well for:

```

- Content generation
- Simple classification tasks
- Basic question answering
- Text transformation

> These models are often more cost-effective for straightforward tasks that don't require complex reasoning.

Reviewer: ```suggestion
These models are often more cost-effective for straightforward tasks that do not require complex reasoning.
```
